### PR TITLE
Add LoggerFactoryOptions.ActivityTrackingOptions

### LINKED ISSUE #21306: Add documentation on how to automatically create a log scope containing the Activity.Current.Id for each request
At the moment the default JSON error response, e.g., for model validation is something like
```json
{
    "errors": {
        "myValue": [
            "The value 'foo' is not valid."
        ]
    },
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
    "title": "One or more validation errors occurred.",
    "status": 400,
    "traceId": "00-1958ab77e30b474b9b3fb63a9644756f-3731b87868e4d345-00"
}
```
This error response includes a `TraceId` which has the value of `Activity.Current.Id`, here `"00-1958ab77e30b474b9b3fb63a9644756f-3731b87868e4d345-00"`.
In order to trace back errors like the one above, I want to add this trace id to all of my log messages.

Is there any way to have the ASP.NET Framework create a log scope for each http request BEFORE passing it on to the controller?
I want to avoid having to wrap each single handler method with
```csharp
using (_logger.BeginScope("{TraceId}", Activity.Current.Id))
{
    ...
}
```
Setting `ConsoleFormatterOptions.IncludeScopes` to `true` does not create the scope for me (obviously). That's only half of the job.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7184a308-a254-9350-a5cd-5bced1f369ae
* Version Independent ID: 726e3bf1-f367-d733-8933-bccc04da0e16
* Content: [Logging in .NET Core and ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0)
* Content Source: [aspnetcore/fundamentals/logging/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/logging/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #21306

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0&branch=pr-en-us-22260#apply-log-filter-rules-in-code)

@svdHero please review

### CODE REVIEW

File: aspnetcore/fundamentals/logging/index.md [hunk 1]
```diff
@@ -897,6 +897,35 @@ The following example shows how to register filter rules in code:
...
+## Automatically log scope with SpanId, TraceId, and ParentId
+
+The logging libraries implicitly create a scope object with `SpanId`, `TraceId`, and `ParentId`. This behavior is configured via <xref:Microsoft.Extensions.Logging.LoggerFactoryOptions.ActivityTrackingOptions>.
+
+```csharp
+  var loggerFactory = LoggerFactory.Create(logging =>
+  {
+      logging.Configure(options =>
+      {
+          options.ActivityTrackingOptions = ActivityTrackingOptions.SpanId
```
Comment (tarekgh) [inline aspnetcore/fundamentals/logging/index.md:913]: In .NET 6.0 we have added extra ActivityTrackingOptions values `Tags` and `Baggage`. do we want to include those too?

Comment (tarekgh) [reply aspnetcore/fundamentals/logging/index.md:913]: Add the following two lines.

```C#
 | ActivityTrackingOptions.Baggage
 | ActivityTrackingOptions.Tags
```

File: aspnetcore/fundamentals/logging/index.md [hunk 2]
```diff
@@ -924,6 +924,33 @@ If the `traceparent` http request header is set, the `ParentId` in the log scope
...
+                                              | ActivityTrackingOptions.TraceId
+                                              | ActivityTrackingOptions.ParentId
+                                              | ActivityTrackingOptions.Baggage
```
Comment (tarekgh) [inline aspnetcore/fundamentals/logging/index.md:941]: This looks good to me. Is there a way to tell 

```
 ActivityTrackingOptions.SpanId
| ActivityTrackingOptions.TraceId
| ActivityTrackingOptions.ParentId
```

are supported on .NET 5.0 and 

```
| ActivityTrackingOptions.Baggage
| ActivityTrackingOptions.Tags;
```

supported on .NET 6.0.

I am seeing `moniker range=">= aspnetcore-6.0"` which the user may think they cannot use SpanId, TraceId, and ParentId on .NET 5.0.

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/logging/index.md:941]: Take a look at [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0&branch=pr-en-us-22260#apply-log-filter-rules-in-code) and use the version selector
Use the version selector:

image

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/logging/index.md:941]: I put two complete versions in the file.

File: aspnetcore/fundamentals/logging/index.md [hunk 3]
```diff
@@ -897,6 +897,62 @@ The following example shows how to register filter rules in code:
...
+          options.IncludeScopes = true;
+      });
+  });
+```
+
+If the `traceparent` http request header is set, the `ParentId` in the log scope shows the W3C `parent-id` from in-bound `traceparent` header and the `SpanId` in the log scope shows the updated `parent-id` for the next out-bound step/span. For more information, see [Mutating the traceparent Field](https://www.w3.org/TR/trace-context/#mutating-the-traceparent-field).
+
+::: moniker-end
+
+::: moniker range=">= aspnetcore-6.0"
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/logging/index.md:927]: .NET 6 and higher. This will become the default version when .NET 6 RTMS.

**Review by tarekgh — APPROVED** [review]

**Pull request merged**